vars/dockerImageExists.groovy

```diff
@@ -0,0 +1,47 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+/**
+  Checks if the given Docker image exists.
+
+  whenTrue(dockerImageExists(image: 'hello-world:latest')) {
+    ...
+  }
+*/
+def call(Map args = [:]) {
+  if (!isInstalled(tool: 'docker', flag: '--version')) {
+    error 'Docker is not installed'
+  }
+
+  def image = args.containsKey('image') ? args.image : error('dockerImageExists: image parameter is required')
+
+  def redirectStdout = isUnix() ? '2>/dev/null' : '2>NUL'
+  if (cmd(returnStatus: true, script: "docker images -q ${image} ${redirectStdout}") == 0) {
```

Comment:
The return code is always 0

```
docker images busybox:foo && echo "It does not work"
```

Reply:
these commands works

```
docker inspect busybox:foo || echo "the image is not here"
```

this will return the Ids if exist locally
```
docker inspect -f "{{.Id}}" busybox:latest
```

Reply:
Ah good catch! Will update the command, thanks!